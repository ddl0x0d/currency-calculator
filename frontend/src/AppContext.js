import React from 'react';

const AppContext = React.createContext({ currencies: [] });

export default AppContext;
