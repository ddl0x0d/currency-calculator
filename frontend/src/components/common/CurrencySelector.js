import AppContext from '../../AppContext';
import Form from 'react-bootstrap/Form';
import PropTypes from 'prop-types';
import React, { useContext } from 'react';

const propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
};

const CurrencySelector = props => {
  const { currencies } = useContext(AppContext);

  return (
    <Form.Group>
      <Form.Label className="small mb-0">{props.label}</Form.Label>
      <Form.Control
        as="select"
        id={props.id}
        size="sm"
        value={props.value}
        onChange={e => props.handleChange(e.target.value)}
      >
        <option key="" value="" />
        {currencies.map(currency => (
          <option key={currency} value={currency}>
            {currency}
          </option>
        ))}
      </Form.Control>
    </Form.Group>
  );
};

CurrencySelector.propTypes = propTypes;

export default CurrencySelector;
