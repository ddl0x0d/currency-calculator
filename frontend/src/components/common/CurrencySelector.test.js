import AppContext from '../../AppContext';
import CurrencySelector from './CurrencySelector';
import { mount, shallow } from 'enzyme';
import React from 'react';

describe('<CurrencySelector />', () => {
  const handleChange = jest.fn().mockName('handleChange');
  const currencySelector = value => (
    <CurrencySelector
      id="currency"
      label="Currency"
      value={value}
      handleChange={handleChange}
    />
  );

  it('renders correctly without currencies', () => {
    const wrapper = mount(currencySelector(''));
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with currencies', () => {
    const component = (
      <AppContext.Provider value={{ currencies: ['EUR', 'GBP'] }}>
        {currencySelector('EUR')}
      </AppContext.Provider>
    );
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles change', () => {
    const wrapper = shallow(currencySelector(''));
    wrapper.find('#currency').simulate('change', { target: { value: 'EUR' } });
    expect(handleChange).toHaveBeenCalledWith('EUR');
  });
});
