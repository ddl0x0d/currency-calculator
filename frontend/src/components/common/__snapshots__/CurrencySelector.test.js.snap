// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencySelector /> renders correctly with currencies 1`] = `
<CurrencySelector
  handleChange={[MockFunction handleChange]}
  id="currency"
  label="Currency"
  value="EUR"
>
  <FormGroup
    as="div"
  >
    <div
      className="form-group"
    >
      <FormLabel
        className="small mb-0"
        column={false}
        srOnly={false}
      >
        <label
          className="small mb-0 form-label"
        >
          Currency
        </label>
      </FormLabel>
      <FormControl
        as="select"
        id="currency"
        onChange={[Function]}
        size="sm"
        value="EUR"
      >
        <select
          className="form-control form-control-sm"
          id="currency"
          onChange={[Function]}
          value="EUR"
        >
          <option
            key=""
            value=""
          />
          <option
            key="EUR"
            value="EUR"
          >
            EUR
          </option>
          <option
            key="GBP"
            value="GBP"
          >
            GBP
          </option>
        </select>
      </FormControl>
    </div>
  </FormGroup>
</CurrencySelector>
`;

exports[`<CurrencySelector /> renders correctly without currencies 1`] = `
<CurrencySelector
  handleChange={[MockFunction handleChange]}
  id="currency"
  label="Currency"
  value=""
>
  <FormGroup
    as="div"
  >
    <div
      className="form-group"
    >
      <FormLabel
        className="small mb-0"
        column={false}
        srOnly={false}
      >
        <label
          className="small mb-0 form-label"
        >
          Currency
        </label>
      </FormLabel>
      <FormControl
        as="select"
        id="currency"
        onChange={[Function]}
        size="sm"
        value=""
      >
        <select
          className="form-control form-control-sm"
          id="currency"
          onChange={[Function]}
          value=""
        >
          <option
            key=""
            value=""
          />
        </select>
      </FormControl>
    </div>
  </FormGroup>
</CurrencySelector>
`;
